### PR TITLE
Update message re beta testing stanc3

### PR DESCRIPTION
**Pull request by ssp3nc3r**
#### Summary:

Change the existing messages to clarify goals of beta testing.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Scott Spencer

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: GPLv3 (http://opensource.org/licenses/GPL-3.0)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: rstan/rstan/R/stanc.R [hunk 1]
```diff
@@ -128,10 +128,12 @@ stanc_beta <- function(model_code, model_name, isystem) {
...
-    message("The NEXT version of Stan will not be able to pre-process your Stan program.\n", 
-            "Please open an issue at\n https://github.com/stan-dev/stanc3/issues \nif you can ",
-            "share or at least describe your Stan program. This will help ensure that Stan\n",
-            "continues to work on your Stan programs in the future. Thank you!\n",
+    message("When you compile models, you are also contributing to development of the NEXT\n",
+            "Stan compiler. In this version of rstan, we compile your model as usual, but\n",
+            "also beta test our next compiler on your syntactically correct model. In this case,\n", 
+            "the anticipated, future compiler didn’t work like we hoped. By submitting the error\n",
+            "as an issue at https://github.com/stan-dev/stanc3/issues, you’ll be contributing\n",
+            "valuable information to the project and ensuring your models keep working. Thank you!\n",
```
Comment (SteveBronder) [inline rstan/rstan/R/stanc.R]: I think I would reword this a little bit just to explain what we mean by "submitting the error"

```suggestion
    message("When you compile models, you are also contributing to development of the NEXT\n",
            "Stan compiler. In this version of rstan, we compile your model as usual, but\n",
            "also test our new compiler on your syntactically correct model. In this case,\n", 
            "the new compiler did not work like we hoped. By filing an issue at\n",
            "https://github.com/stan-dev/stanc3/issues with your model\n",
            "or a minimal example that shows this warning you will be contributing\n",
            "valuable information to Stan and ensuring your models continue working.",
            " Thank you!\n",
```

**Review by jgabry — APPROVED** [review]

**Comment by jgabry** [comment]
@bgoodri I fixed merge conflicts so this is ready to go but I'll let you merge it in case you want to make any changes to the message.

**Comment by bgoodri** [comment]
Go ahead
